tensorflow/core/util/gpu_kernel_helper.h

```diff
@@ -13,34 +13,76 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 
-#ifndef TENSORFLOW_CORE_UTIL_CUDA_KERNEL_HELPER_H_
-#define TENSORFLOW_CORE_UTIL_CUDA_KERNEL_HELPER_H_
+#ifndef TENSORFLOW_CORE_UTIL_GPU_KERNEL_HELPER_H_
+#define TENSORFLOW_CORE_UTIL_GPU_KERNEL_HELPER_H_
 
-#if GOOGLE_CUDA
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 
-#include "tensorflow/core/util/cuda_device_functions.h"
-#include "tensorflow/core/util/cuda_launch_config.h"
+#include "tensorflow/core/util/gpu_device_functions.h"
+#include "tensorflow/core/util/gpu_launch_config.h"
 
+#if GOOGLE_CUDA
 #include "cuda/include/cuda_fp16.h"
+#endif
+
+#if GOOGLE_CUDA
+#define TF_RED_WARPSIZE 32
+#elif TENSORFLOW_USE_ROCM
+#define TF_RED_WARPSIZE 64
+#endif
+
+#if GOOGLE_CUDA
+#define GPU_LAUNCH_KERNEL(kernel, block_count, threads_per_block, \
```

Comment:
Is this still needed?

Reply:
This is one of the most important feature in this PR. Quite a few TF operators which depends on CUDA-specific `<<< >>>` kernel launch syntax would be changed to this macro.

Granted, I see several operators now moving toward using `CudaLaunchKernel` but still some of them employ `<<< >>>` therefore this macro is needed.

Reply:
Ah, I see. Sounds good then. I will work towards using CudaLaunchKernel everywhere, but this is fine for now.